projects/WTI-UI/src/app/modules/core/services/contest.service.ts

```diff
@@ -74,5 +117,100 @@ export class ContestService extends IContestService {
 	getStandingsAreCurrentFlag() : boolean {
 		return this.standingsAreCurrent ;
 	}
+	
+	/** This method invokes the local getContestClock() method, which makes an HTTP call to the WTI-API to get the current
+	 *  PC2 Server clock (aka "ContestTime").  It subscribes to the Observable returned by the HTTP call, 
+	 *  and when the subscription callback occurs it uses the received ContestClock data (an instance of 
+	 *  WTI=UI models/ContestClock) to update the WTI-UI contest clock (including the onscreen displays).  
+	 */
+	updateLocalContestClockFromServer ()  {
+		
+		//get the actual contest clock info from the PC2 server via the Contest Service (which gets it via the WTI Server and its PC2 API)
+		if (DEBUG_MODE) {
+			console.log("  Invoking ContestService.getContestClock(), subscribing for HTTP callback result")
+		}
+		this.getContestClock()
+			.subscribe(
+				(data: any) => {
+        			if (!data) { 
+						console.error ("ContestService.updateLocalContestClockFromServer() getContestClock() subscription callback: unable to get ContestClock from PC2 API via ContestService!");
+					} else {
+						if (DEBUG_MODE) {
+							console.log("ContestService.updateLocalContestClockFromServer(): got callback from getContestClock() subscription.");
+						}
+						//install the received contest clock data into the ContestService's ContestClock
+						this.installNewContestClock(data);					
+					}
+      			}, 
+				(error: any) => {
+        			console.error("ContestService.updateLocalContestClockFromServer(): getContestClock() subscription callback error: ");
+					console.error (error);
+      			}
+			);	
+	}
+	
+	/** This method receives a WTI-UI ContestClock model containing new values which should be used to update the WTI-UI contest clock,
+	 *  including the onscreen displays.  It constructs a new ContestClock object containing the received data and installs that
+	 *  object as the current WTI-UI clock.  It then updates the separate "ContestTimer" object with the specified values, and
+	 *  if the received data indicates the clock should be running it starts the ContestTimer (which then genrates a "clock tick"
+	 *  once per second to update the clock displays).
+	 */
+	installNewContestClock(data: any) {
+		
+		//copy the data fields (received from the PC2 Server via the WTI-API) into a new ContestService ContestClock object
+		let newContestClock = new ContestClock();
+		newContestClock.isRunning = data.running ;
+		newContestClock.contestLengthSecs = data.contestLengthInSecs ;
+		newContestClock.elapsedSecs = data.elapsedSecs ;
+		newContestClock.wallClockStartTime = data.wallClockStartTime ;
 
+		//save the new clock
+		if (DEBUG_MODE) {
+			console.log("ContestService (id", this.uniqueId, ").installNewContestClock(): replacing Contest Clock with:");
+			console.log(newContestClock);
+		}
+		this.contestClock = newContestClock;
+		
+		//pull the values out of the updated clock
```

Comment:
Similar comment on usage of "let"

Reply:
And similar reply; I agree and I'll change them all to `const`.

Reply:
Fixed in next push.